### PR TITLE
Store PIP_INDEX_URL value in pip.ini config when present during assemble

### LINKED ISSUE #654: Use PIP_INDEX_URL with pipenv
`pipenv` does not read the `PIP_INDEX_URL` environment variable (see: https://github.com/pypa/pipenv/issues/4808 ) so to when using a local pip mirror I have to define the mirror url twice, both with `PIP_INDEX_URL` env var (for installing `pipenv`) and also in the `Pipfile`.
While it's not a huge problem it would still be nice to be able to configure the mirror with a single env var.

I see two possible solutions:

To either pass the value of `PIP_INDEX_URL` to the `pipenv install` command with the `--index` option
```   
pipenv install --deploy --index $PIP_INDEX_URL
```

or write the `PIP_INDEX_URL` env var to a `pip.conf` or `pip.ini` during the assemble.

### PR DESCRIPTION
**Pull request by hrnciar**
Fixes #654

### CODE REVIEW

**Comment by hrnciar** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test-all]

File: test/run [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # IMAGE_NAME specifies a name of the candidate image used for testing.
 # The image has to be available before this script is executed.
 #
-declare -a COMMON_WEB_APPS=({gunicorn-config-different-port,gunicorn-different-port,django-different-port,standalone,setup,setup-requirements,django,numpy,app-home,locale,pipenv,pipenv-and-micropipenv-should-fail,app-module,pyuwsgi-pipenv{% if spec.version.startswith("3.") %},micropipenv{% endif %}}-test-app)
+declare -a COMMON_WEB_APPS=({gunicorn-config-different-port,gunicorn-different-port,django-different-port,standalone,standalone-custom-pypi-index-test-app,setup,setup-requirements,django,numpy,app-home,locale,pipenv,pipenv-and-micropipenv-should-fail,app-module,pyuwsgi-pipenv{% if spec.version.startswith("3.") %},micropipenv{% endif %}}-test-app)
```
Comment (frenzymadness) [inline test/run:9]: This list uses a shell expansion so the name of the test should not contain `-test-app`. See the rest of the tests in the list.

**Comment by hrnciar** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test]

**Comment by phracek** [comment]
[test-all]

**Comment by hrnciar** [comment]
[test-all]

**Comment by phracek** [comment]
RHEL8 - Python 2.7 failure:

```
[1m[INFO] Building the standalone-custom-pypi-index-test-app application image ...[0m

STEP 1/10: FROM ubi8/python-27:2.7
STEP 2/10: LABEL "io.openshift.s2i.build.image"="ubi8/python-27:2.7"       "io.openshift.s2i.build.source-location"="file:///root/sclorg/s2i-python-container/2.7/test/standalone-custom-pypi-index-test-app"
--> c24fa516179e
STEP 3/10: USER root
--> a85aad6bcd55
STEP 4/10: COPY upload/src/ /tmp/src
--> 2a56e8ea3a8d
STEP 5/10: RUN chown -R 1001:0 /tmp/src
--> 8d8bf7393fc1
STEP 6/10: ENV PIP_INDEX_URL=https://test.pypi.org/simple/
--> 87f8e451f591
STEP 7/10: ENV DISABLE_SETUP_PY_PROCESSING=true
--> a37dfc85b43d
STEP 8/10: USER 1001
--> f5ff008efcd2
STEP 9/10: RUN /usr/libexec/s2i/assemble
---> Installing application source ...
---> Upgrading pip to latest version supported for Python 2.7 ...
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Looking in indexes: https://test.pypi.org/simple/
Requirement already up-to-date: pip in /opt/app-root/lib/python2.7/site-packages (20.3.4)
---> Creating pip.ini config file
---> Installing dependencies ...
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Looking in indexes: https://test.pypi.org/simple/
Collecting pello
  Downloading https://test-files.pythonhosted.org/packages/e6/51/fbdfb3630b042e3f2c1afe5a6e1a9ed5ba11c48640cb6248c91ed44a3b4e/Pello-1.0.3.tar.gz (5.9 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /opt/app-root/bin/python2 /opt/app-root/lib/python2.7/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-t3SzWE/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://test.pypi.org/simple/ -- 'setuptools>=42' wheel
       cwd: None
  Complete output (4 lines):
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
  Looking in indexes: https://test.pypi.org/simple/
  ERROR: Could not find a version that satisfies the requirement setuptools>=42 (from versions: none)
  ERROR: No matching distribution found for setuptools>=42
  ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/app-root/bin/python2 /opt/app-root/lib/python2.7/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-t3SzWE/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://test.pypi.org/simple/ -- 'setuptools>=42' wheel Check the logs for full command output.
Error: building at STEP "RUN /usr/libexec/s2i/assemble": while running runtime: exit status 1
  ERROR: Failed to to build /tmp/tmp.ENahEOxKAc/Dockerfile.Com3
```

**Comment by hrnciar** [comment]
[test]

**Comment by frenzymadness** [comment]
The PR looks good to me now. Thank you.

@gardar do you want to take a look and test it before we merge it and ship it?

**Comment by frenzymadness** [comment]
[test]

**Comment by frenzymadness** [comment]
[test]

**Comment by gardar** [comment]
> @gardar do you want to take a look and test it before we merge it and ship it?

Haven't had a chance to test it, but it looks good to me, thanks!

**Pull request merged**